cdc/generation.cc

```diff
@@ -345,39 +345,400 @@ static void do_update_streams_description(
 
     std::vector<cdc::stream_id> streams_vec(streams_set.begin(), streams_set.end());
 
-    sys_dist_ks.create_cdc_desc(streams_ts, streams_vec, ctx).get();
+    co_await sys_dist_ks.create_cdc_desc(streams_ts, streams_vec, ctx);
     cdc_log.info("CDC description table successfully updated with generation {}.", streams_ts);
 }
 
-void update_streams_description(
+future<> update_streams_description(
         db_clock::time_point streams_ts,
         shared_ptr<db::system_distributed_keyspace> sys_dist_ks,
         noncopyable_function<unsigned()> get_num_token_owners,
         abort_source& abort_src) {
     try {
-        do_update_streams_description(streams_ts, *sys_dist_ks, { get_num_token_owners() });
-    } catch(...) {
+        co_await do_update_streams_description(streams_ts, *sys_dist_ks, { get_num_token_owners() });
+    } catch (...) {
         cdc_log.warn(
             "Could not update CDC description table with generation {}: {}. Will retry in the background.",
             streams_ts, std::current_exception());
 
         // It is safe to discard this future: we keep system distributed keyspace alive.
-        (void)seastar::async([
-            streams_ts, sys_dist_ks, get_num_token_owners = std::move(get_num_token_owners), &abort_src
-        ] {
+        (void)(([
+            _streams_ts = streams_ts,
+            _sys_dist_ks = std::move(sys_dist_ks),
+            _get_num_token_owners = std::move(get_num_token_owners),
+            &_abort_src = abort_src
+        ] () mutable -> future<> {
+            auto streams_ts = _streams_ts;
+            auto sys_dist_ks = std::move(_sys_dist_ks);
+            auto get_num_token_owners = std::move(_get_num_token_owners);
+            auto& abort_src = _abort_src;
+
```

Comment:
The lambda must capture something.
So if I understand correctly, I could capture everything by ref using `[&]` and then move the stuff. Because the lambda is immediately invoked, the moves happen before the caller's frame gets freed. Is that right?

Reply:
Yes. That should be the case. As you point out, it does not really change anything, performancewise or so, just maybe shorter to write and read. And does not give a false impression that the lambda actually has state. As said, super nitpicky, not very important.